zml/tokenizer.zig

```diff
@@ -144,7 +199,18 @@ pub const Tokenizer = struct {
         var num_tokens: usize = 0;
         var off: usize = 0;
         while (off < input.len) {
-            const utf_len = try std.unicode.utf8ByteSequenceLength(input[off]);
+            var utf_len: usize = try std.unicode.utf8ByteSequenceLength(input[off]);
+            if (utf_len == 1 and input[off] == '<') {
```

Comment:
what is this code doing ?

also it's not a good idea to always preprocess text like this. Because the user query should not be allowed to introduce special tokens.

Reply:
This code is checking whether the token is potentially a hex value (e.g. `<0x0A>`) and reads ahead to prevent this token from being split/merged.